rbd: check for clusterid mapping in RegenerateJournal()

Rakshith-R opened the pull request:
- rbd: move GetMappedID() to util package
and 

- rbd: check for clusterid mapping in RegenerateJournal()

This commit adds fetchMappedClusterIDAndMons() which returns
monitors and clusterID info after checking cluster mapping info.
This is required for regenerating omap entries in mirrored cluster
with different clusterID.

Signed-off-by: Rakshith R <rar@redhat.com>

Tested with mapping clusterid "rook-ceph":"rook-ceph-1"
With no corresponding mon ip list for "rook-ceph"
Works ! 
```
I0826 11:50:58.585164       1 rbd_util.go:1175] found new clusterID mapping "rook-ceph-1" for existing clusterID "rook-ceph"
E0826 11:50:58.588179       1 omap.go:78] omap not found (pool="replicapool", namespace="", name="csi.volumes.default"): rados: ret=-2, No such file or directory
I0826 11:50:58.621629       1 omap.go:155] set omap keys (pool="replicapool", namespace="", name="csi.volumes.default"): map[csi.volume.pvc-278e6845-46a3-4603-97d4-272b3bbe6747:3c742b18-0663-11ec-88bd-0242ac110005])
I0826 11:50:58.633764       1 omap.go:155] set omap keys (pool="replicapool", namespace="", name="csi.volume.3c742b18-0663-11ec-88bd-0242ac110005"): map[csi.imagename:csi-vol-3c742b18-0663-11ec-88bd-0242ac110005 csi.volname:pvc-278e6845-46a3-4603-97d4-272b3bbe6747 csi.volume.owner:rook-ceph])
I0826 11:50:58.633806       1 rbd_journal.go:641] re-generated Volume ID (0001-000b-rook-ceph-1-0000000000000001-3c742b18-0663-11ec-88bd-0242ac110005) and image name (csi-vol-3c742b18-0663-11ec-88bd-0242ac110005) for request name (pvc-278e6845-46a3-4603-97d4-272b3bbe6747)
I0826 11:50:58.689962       1 omap.go:155] set omap keys (pool="replicapool", namespace="", name="csi.volume.3c742b18-0663-11ec-88bd-0242ac110005"): map[csi.imageid:12955994250b])
I0826 11:50:58.690234       1 rbd_util.go:1175] found new clusterID mapping "rook-ceph-1" for existing clusterID "rook-ceph"
I0826 11:50:58.693026       1 omap.go:87] got omap values: (pool="replicapool", namespace="", name="csi.volumes.default"): map[csi.volume.pvc-278e6845-46a3-4603-97d4-272b3bbe6747:3c742b18-0663-11ec-88bd-0242ac110005]
I0826 11:50:58.693548       1 omap.go:87] got omap values: (pool="replicapool", namespace="", name="csi.volume.3c742b18-0663-11ec-88bd-0242ac110005"): map[csi.imageid:12955994250b csi.imagename:csi-vol-3c742b18-0663-11ec-88bd-0242ac110005 csi.volname:pvc-278e6845-46a3-4603-97d4-272b3bbe6747 csi.volume.owner:rook-ceph]

```

Rakshith-R committed:
rbd: check for clusterid mapping in RegenerateJournal()

This commit adds fetchMappedClusterIDAndMons() which returns
monitors and clusterID info after checking cluster mapping info.
This is required for regenerating omap entries in mirrored cluster
with different clusterID.

Signed-off-by: Rakshith R <rar@redhat.com>

nixpanic commented on internal/rbd/rbd_util.go:
```diff
@@ -1159,6 +1159,42 @@ func getMappedID(key, value, id string) string {
 	return ""
 }
 
+// fetchMappedClusterIDAndMons returns monitors and clusterID info after checking cluster mapping.
```
Please add a unit test.

Probably good to move this to `internal/util/cluster_mapping.go` as it is not rbd specific?

Rakshith-R replied:
>Please add a unit test.
Probably good to move this to internal/util/cluster_mapping.go as it is not rbd specific?

Moved it to util package and added unit tests, PTAL @nixpanic @humblec @ceph/ceph-csi-contributors

Rakshith-R committed:
rbd: move GetMappedID() to util package

This commit moves getMappedID() from rbd to util
package since it is not rbd specific and exports
it from there.

Signed-off-by: Rakshith R <rar@redhat.com>

Rakshith-R commented:
@Mergifyio rebase

Rakshith-R commented:
Adding backport to release v3.4 label.
